skrub/datasets/_utils.py

```diff
@@ -1,8 +1,38 @@
-import os
 from pathlib import Path
 
 
-def get_data_dir(name: str | None = None) -> Path:
+def get_data_home(data_home: Path | None = None) -> Path:
+    """Returns the path of the skrub data directory.
+
+    This folder is used by some large dataset loaders to avoid downloading the
+    data several times.
+
+    By default the data directory is set to a folder named 'skrub_data' in the
+    user home folder.
+
+    Alternatively, it can be set programmatically by giving an explicit folder
+    path. The '~' symbol is expanded to the user home folder.
+
+    If the folder does not already exist, it is automatically created.
+
+    Parameters
+    ----------
+    data_home : Path, default=None
+        The path to skrub data directory. If `None`, the default path
+        is `~/skrub_data`.
+
+    Returns
+    -------
+    data_home: str or path-like, default=None
+        The path to skrub data directory.
+    """
+    if data_home is None:
+        data_home = Path("~") / "skrub_data"
+    data_home.mkdir(parents=True, exist_ok=True)
```

Comment:
Maybe we want to be more rigid on this, like, I don't think we want to create the user folder if it doesn't already exist. Or do we?

Reply:
In scikit-learn, we do create the folder. It is a bit annoying to get an error and have to create by hand a folder when calling the fetcher the first time.